data/public_data/controller.py

```diff
@@ -76,21 +101,34 @@ def update_movie_data(self, lower, upper, delay):
                 logging.error(e)
                 logging.error(current_imdb_id)
 
-        logging.warning("Movie data update process complete.")
-
-    def update_movie_rating(self):
+    def _update_movie_rating(self, movie_ids):
         """
-        updates movie rating from various websites
+        updates movie rating for a list of movie ids
+        from various websites
+        :param movie_ids: list
+        :return:
         """
-        logging.warning("Initialise movie rating update process ...")
-
-        existing_movies_id = self.loader.get_movie_id_list()
-        for current_id in existing_movies_id:
-            self._update_single_movie_rating(current_id)
+        logging.warning("Initialise movie rating retrieval process ...")
+        for current_imdb_id in movie_ids:
+            try:
+                self._update_single_movie_rating(current_imdb_id)
+            except error.HTTPError:  # invalid id will cause an 404 error
+                continue
+            except ConnectionResetError or TimeoutError or client.IncompleteRead:
+                logging.error("Connection reset by remote host, reconnecting in 5s ...")
+                time.sleep(5)
 
-        logging.warning("Movie rating update process complete.")
+                # try again
+                try:
+                    self._update_single_movie_data(current_imdb_id)
+                except:  # skip any error
+                    continue
+            except Exception as e:  # unknown error
+                logging.error("Unknown error occurs. Please examine.")
```

Comment:
This clause is used because I need to run the script and adding potential new error handling clauses.